sympy/physics/mechanics/joint.py

```diff
@@ -543,3 +533,217 @@ def _set_linear_velocity(self):
+    kdes : list
+        Kinematical differential equations of the joint.
+
+    Examples
+    =========
+
+    A single prismatic joint is created from two bodies and has the following basic
+    attributes:
+
+    >>> from sympy.physics.mechanics import Body, PrismaticJoint
+    >>> parent = Body('P')
+    >>> parent
+    P
+    >>> child = Body('C')
+    >>> child
+    C
+    >>> joint = PrismaticJoint('PC', parent, child)
+    >>> joint
+    PrismaticJoint: PC  parent: P  child: C
+    >>> joint.name
+    'PC'
+    >>> joint.parent
+    P
+    >>> joint.child
+    C
+    >>> joint.parent_point
+    PC_P_joint
+    >>> joint.child_point
+    PC_C_joint
+    >>> joint.parent_axis
+    P_frame.x
+    >>> joint.child_axis
+    C_frame.x
+    >>> joint.coordinates
+    [x_PC(t)]
+    >>> joint.speeds
+    [v_PC(t)]
+    >>> joint.child.frame.ang_vel_in(joint.parent.frame)
+    0
+    >>> joint.child.frame.dcm(joint.parent.frame)
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1]])
+    >>> joint.child_point.pos_from(joint.parent_point)
+    x_PC(t)*P_frame.x
+
+    To further demonstrate the use of the prismatic joint, the kinematics of n-link
+    pendulum(n=2) about the X axis of each connected body can be created as follows.
+
+    >>> from sympy.physics.mechanics import PrismaticJoint, Body
+
+    First create bodies to represent the fixed ceiling and one to represent
+    a particle.
+
+    >>> ceiling = Body('C')
+    >>> Part1 = Body('P1')
+    >>> Part2 = Body('P2')
+
+    The first joint will connect the particle to the ceiling and the
+    joint axis will be about the X axis for each body.
+
+    >>> J1 = PrismaticJoint('J1', ceiling, Part1)
+
+    The second joint will connect the second particle to the first particle
+    and the joint axis will also be about the X axis for each body.
+
+    >>> J2 = PrismaticJoint('J1', Part1, Part2)
+
+    Once the joint is established the kinematics of the connected bodies can
+    be accessed. First the direction cosine matrices of Part relative
+    to the ceiling are found:
+
+    >>> Part1.dcm(ceiling)
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1]])
+
+    >>> Part2.dcm(ceiling)
+    Matrix([
+    [1, 0, 0],
+    [0, 1, 0],
+    [0, 0, 1]])
+
+    The position of the particles' masscenter is found with:
+
+    >>> Part1.masscenter.pos_from(ceiling.masscenter)
+    x_J1(t)*C_frame.x
+
+    >>> Part2.masscenter.pos_from(ceiling.masscenter)
+    x_J1(t)*C_frame.x + x_J1(t)*P1_frame.x
+
+    The angular velocities of the two particle links can be computed with
+    respect to the ceiling.
+
+    >>> Part1.ang_vel_in(ceiling)
+    0
+
+    >>> Part2.ang_vel_in(ceiling)
+    0
+
+    And finally, the linear velocities of the two particles can be computed
+    with respect to the ceiling.
+
+    >>> Part1.masscenter_vel(ceiling)
+    v_J1(t)*C_frame.x
+
+    >>> Part2.masscenter_vel(ceiling)
+    v_J1(t)*C_frame.x + Derivative(x_J1(t), t)*P1_frame.x
```

Comment:
This is supposed to be:

```
v_J1(t)*C_frame.x + v_J2(t)*P1_frame.x
```

Reply:
Bur derivative (x) = v

Reply:
Yes, but the joint's should ensure that the generalized speeds are used. We shouldn't see `Derivatie(coordinate, t)` in any of the kinematics.

Reply:
Also, it says `Derivative(x_J1(t), t)` J1 is not correct either.

Reply:
> Also, it says `Derivative(x_J1(t), t)` J1 is not correct either.

Yes , I named joint 2 'J1' by mistake.

Reply:
> Yes, but the joint's should ensure that the generalized speeds are used. We shouldn't see `Derivatie(coordinate, t)` in any of the kinematics.

I agree, but how can we do that, part2 velocity is defined in part1's frame, we want it's velocity in C's frame.
The current velocity function returns the `velovity of part1 in C + derivative(pos_vector between part1 and part2)`. So how can we change this behaviour?

Reply:
PR #21730 might solve this issue.

Reply:
All joints should be setting the correctly velocities, no computing them.

Reply:
For example, if you are doing a `pos_vec.diff(t)` in your joint code, that's not the way to do it. The joint is defined and we know what the velocity is, so just use `.set_vel()` to set the correct velocities.

Reply:
We are set velocity for parent frame, but we can't for parent body' parent body.  So we have to compute it for that.

Reply:
Fixed!